rero_ils/query.py

```diff
@@ -130,6 +130,42 @@ def loans_search_factory(self, search, query_parser=None):
     return (search, urlkwargs)
 
 
+def patron_transactions_search_factory(self, search, query_parser=None):
+    """Patron transaction search factory.
+
+    Restricts results to oraganisation level for librarian and sys_lib.
+    Restricts results to his transactions for users with role patron.
+    """
+    search, urlkwargs = search_factory(self, search)
+    if current_patron:
+        if current_patron.is_librarian or current_patron.is_system_librarian:
+            search = search.filter(
+                'term', organisation__pid=current_patron.get_organisation(
+                    )['pid'])
+        if current_patron.is_patron:
+            search = search.filter(
+                'term', patron_pid=current_patron.pid)
+    return (search, urlkwargs)
+
+
+def patron_transaction_events_search_factory(self, search, query_parser=None):
+    """Patron transaction events search factory.
+
+    Restricts results to oraganisation level for librarian and sys_lib.
+    Restricts results to his transactions for users with role patron.
+    """
+    search, urlkwargs = search_factory(self, search)
+    if current_patron:
+        if current_patron.is_librarian or current_patron.is_system_librarian:
+            search = search.filter(
+                'term', organisation__pid=current_patron.get_organisation(
+                    )['pid'])
+        if current_patron.is_patron:
+            search = search.filter(
+                'term', patron_pid=current_patron.pid)
+    return (search, urlkwargs)
```

Comment:
Both methods are the same. Maybe only used a generic one

Reply:
fixed